src/napari_omero/plugins/loaders.py

```diff
@@ -65,52 +65,57 @@ def omero_proxy_reader(
 
 
 def load_image_wrapper(image: ImageWrapper) -> List[LayerData]:
-    return [
-        load_omero_channel(image, channel, c)
-        for c, channel in enumerate(image.getChannels())
+    data = get_data_lazy(image)
+    meta = get_omero_metadata(image)
+    # contrast limits range ... not accessible from plugin interface
+    # win_min = channel.getWindowMin()
+    # win_max = channel.getWindowMax()
+    return [(data, meta)]
+
+
+def get_omero_metadata(image: ImageWrapper) -> Dict:
+    """Get metadata from OMERO as a Dict to pass to napari."""
+    channels = image.getChannels()
+
+    colors = []
+    for ch in channels:
+        # use current rendering settings from OMERO
+        color = ch.getColor().getRGB()
+        color = [r / 256 for r in color]
+        colors.append(Colormap([[0, 0, 0], color]))
+
+    contrast_limits = [
+        [ch.getWindowStart(), ch.getWindowEnd()] for ch in channels
     ]
 
+    visibles = [ch.isActive() for ch in channels]
+    names = [ch.getLabel() for ch in channels]
 
-def load_omero_channel(
-    image: ImageWrapper, channel: ChannelWrapper, c_index: int
-) -> LayerData:
-    data = get_data_lazy(image, c_index=c_index)
-    color = channel.getColor().getRGB()
-    color = [r / 256 for r in color]
-    cmap = Colormap([[0, 0, 0], color])
     scale = None
-
-    # FIXME: still getting size mismatch sometimes  is there a getNDim()?
     if image.getSizeZ() > 1:
         size_x = image.getPixelSizeX()
         size_z = image.getPixelSizeZ()
         if size_x is not None and size_z is not None:
-            if image.getSizeT() > 1:
-                scale = [1, size_z / size_x, 1, 1]
-            else:
-                scale = [size_z / size_x, 1, 1]
-
-    meta = {
-        "blending": "additive",
-        "colormap": ("from_omero", cmap),
-        "scale": scale,
-        "name": channel.getLabel(),
-        "visible": channel.isActive(),
-        "contrast_limits": [channel.getWindowStart(), channel.getWindowEnd()],
+            scale = [1, size_z / size_x, 1, 1]
+
+    return {
+        'channel_axis': 1,
+        'colormap': colors,
+        'contrast_limits': contrast_limits,
+        'name': names,
+        'visible': visibles,
+        'scale': scale,
```

Comment:
while we're at it, we might want to add something to the `{'metadata': ... }` field (in the met dict).  That's a place where devs can put anything they want... we could drop the actual wrapper object there?  or perhaps just the proxy string like "Image:1"?  
Ultimately, I'm thinking down the road about saving ROIs and stuff back to the server.  That's easy when we know we only have a single image open... but if we eventually allow multiple images to be open, we'll need to know which image the ROI corresponds to, and retaining some of the OMERO metadata will make it easier to know what our "options" are.  thoughts?

Reply:
Created https://github.com/tlambert03/napari-omero/issues/16 so we don't have to decide before merging this